runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java

```diff
@@ -638,6 +651,37 @@ private boolean checkAnimationIdle(Duration waitDuration, Player local)
 		return false;
 	}
 
+	private boolean checkMovementIdle(Duration waitDuration, Player local)
+	{
+		if (lastPosition == null)
+		{
+			lastPosition = local.getWorldLocation();
+			return false;
+		}
+
+		WorldPoint position = local.getWorldLocation();
+
+		if (lastPosition.equals(position))
+		{
+			if (notifyPosition
+				&& local.getAnimation() == IDLE
+				&& Instant.now().compareTo(lastMoving.plus(waitDuration)) >= 0)
+			{
+				notifyPosition = false;
+				// Return true only if we weren't just breaking out of an animation
```

Comment:
You could maybe elaborate on this? I think this is so if there is another idle notification pending due to the animation it will instead defer to that?

Reply:
Yes that's right. 

This is to cover the case where a player makes an interaction (from a distance) that also starts an animation. So if a player chops a tree, and the tree falls, the idle movement notification won't fire.